spyder_kernels/console/kernel.py

```diff
@@ -705,15 +706,39 @@ def set_traceback_highlighting(self, color_scheme):
 
     def set_traceback_syntax_highlighting(self, syntax_style):
         """Set the traceback syntax highlighting style."""
-        import IPython.core.ultratb
-        from IPython.core.ultratb import VerboseTB
+        if parse_version(ipython_release.version) < parse_version("9.0"):
+            # Use `tb_highlight_style` class attribute to set the style
+            import IPython.core.ultratb
+            from IPython.core.ultratb import VerboseTB
+
+            IPython.core.ultratb.get_style_by_name = create_style_class
 
-        IPython.core.ultratb.get_style_by_name = create_style_class
+            if getattr(VerboseTB, 'tb_highlight_style', None) is not None:
+                VerboseTB.tb_highlight_style = syntax_style
+            elif getattr(VerboseTB, '_tb_highlight_style', None) is not None:
+                VerboseTB._tb_highlight_style = syntax_style
+        else:
+            # Create theme definition and set it as `_default`
+            import IPython.utils.PyColorize
+            from IPython.utils.PyColorize import (
+                Theme,
+                linux_theme,
+                lightbg_theme,
+            )
 
-        if getattr(VerboseTB, 'tb_highlight_style', None) is not None:
-            VerboseTB.tb_highlight_style = syntax_style
-        elif getattr(VerboseTB, '_tb_highlight_style', None) is not None:
-            VerboseTB._tb_highlight_style = syntax_style
+            extra_style = (
+                linux_theme.extra_style
+                if self.shell.get_spyder_theme() == "dark"
+                else lightbg_theme.extra_style
+            )
+            extra_style.update(create_pygments_dict(syntax_style))
+            theme = Theme(
+                "custom_theme",
+                None,
+                extra_style,
+            )
+            IPython.utils.PyColorize.theme_table["custom_theme"] = theme
+            self.shell.run_line_magic("colors", "custom_theme")
```

Comment:
Create theme as `spyder_theme` instead of `custom_theme`:
```suggestion
            extra_style.update(create_pygments_dict(syntax_style))
            theme = Theme(
                "spyder_theme",
                None,
                extra_style,
            )
            IPython.utils.PyColorize.theme_table["spyder_theme"] = theme
            self.shell.run_line_magic("colors", "spyder_theme")
```